src/borg/testsuite/__init__.py

```diff
@@ -256,6 +256,17 @@ def wait_for_mount(self, path, timeout=5):
             time.sleep(.1)
         raise Exception('wait_for_mount(%s) timeout' % path)
 
+    @contextmanager
+    def read_only(self, path):
+        """Some paths need to be made read-only for testing"""
+        try:
+            # Using chmod is not enough due to root privileges
+            os.system('chattr +i "%s"' % path)
```

Comment:
this only works if OS and FS supports this.

maybe check result code of the call?

Reply:
> maybe check result code of the call?

:+1: Added an `NotImplementedError`.